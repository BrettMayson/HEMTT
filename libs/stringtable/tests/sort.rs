#![allow(clippy::unwrap_used)]

use hemtt_stringtable::Project;
use hemtt_workspace::WorkspacePath;

#[test]
fn sort_ace_arsenal() {
    let mut stringtable =
        Project::read(WorkspacePath::slim_file("tests/ace_arsenal.xml").unwrap()).unwrap();
    insta::assert_debug_snapshot!(stringtable);

    stringtable.sort();

    insta::assert_debug_snapshot!(stringtable);
}

#[test]
fn sort_comments() {
    let mut stringtable =
        Project::read(WorkspacePath::slim_file("tests/sort/comments.xml").unwrap()).unwrap();
    insta::assert_debug_snapshot!(stringtable);

    stringtable.sort();

    insta::assert_debug_snapshot!(stringtable);

    let mut out = String::new();
    stringtable.to_writer(&mut out, false).unwrap();

    insta::assert_snapshot!(out);
}

#[test]
fn sort_gh822() {
    let mut stringtable =
        Project::read(WorkspacePath::slim_file("tests/sort/gh822.xml").unwrap()).unwrap();
    stringtable.sort();

    insta::assert_debug_snapshot!(stringtable);

    let mut out = String::new();
    stringtable.to_writer(&mut out, false).unwrap();

    insta::assert_snapshot!(out);
}

#[test]
fn sort_containers() {
    let mut stringtable =
        Project::read(WorkspacePath::slim_file("tests/sort/containers.xml").unwrap()).unwrap();
    stringtable.sort();

    insta::assert_debug_snapshot!(stringtable);

    let mut out = String::new();
    stringtable.to_writer(&mut out, false).unwrap();

    insta::assert_snapshot!(out);
}
