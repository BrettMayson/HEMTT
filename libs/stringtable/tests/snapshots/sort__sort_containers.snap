---
source: libs/stringtable/tests/sort.rs
expression: stringtable
---
Project {
    inner: InnerProject {
        name: "test",
        packages: [
            Package {
                name: "test",
                keys: [
                    Key {
                        id: "A1",
                        original: Some(
                            "Origin",
                        ),
                        english: None,
                        czech: None,
                        french: None,
                        spanish: None,
                        italian: None,
                        polish: None,
                        portuguese: None,
                        russian: None,
                        german: None,
                        korean: None,
                        japanese: None,
                        chinese: None,
                        chinesesimp: None,
                        turkish: None,
                        swedish: None,
                        slovak: None,
                        serbocroatian: None,
                        norwegian: None,
                        icelandic: None,
                        hungarian: None,
                        greek: None,
                        finnish: None,
                        dutch: None,
                        ukrainian: None,
                        danish: None,
                    },
                ],
                containers: [
                    Package {
                        name: "A",
                        keys: [
                            Key {
                                id: "A3",
                                original: Some(
                                    "Origin",
                                ),
                                english: None,
                                czech: None,
                                french: None,
                                spanish: None,
                                italian: None,
                                polish: None,
                                portuguese: None,
                                russian: None,
                                german: None,
                                korean: None,
                                japanese: None,
                                chinese: None,
                                chinesesimp: None,
                                turkish: None,
                                swedish: None,
                                slovak: None,
                                serbocroatian: None,
                                norwegian: None,
                                icelandic: None,
                                hungarian: None,
                                greek: None,
                                finnish: None,
                                dutch: None,
                                ukrainian: None,
                                danish: None,
                            },
                        ],
                        containers: [],
                    },
                    Package {
                        name: "B",
                        keys: [
                            Key {
                                id: "A2",
                                original: None,
                                english: Some(
                                    "EN",
                                ),
                                czech: None,
                                french: None,
                                spanish: None,
                                italian: None,
                                polish: None,
                                portuguese: None,
                                russian: None,
                                german: Some(
                                    "DE",
                                ),
                                korean: None,
                                japanese: None,
                                chinese: None,
                                chinesesimp: None,
                                turkish: None,
                                swedish: None,
                                slovak: None,
                                serbocroatian: None,
                                norwegian: None,
                                icelandic: None,
                                hungarian: None,
                                greek: None,
                                finnish: None,
                                dutch: None,
                                ukrainian: None,
                                danish: None,
                            },
                            Key {
                                id: "B1",
                                original: Some(
                                    "Origin",
                                ),
                                english: None,
                                czech: None,
                                french: None,
                                spanish: None,
                                italian: None,
                                polish: None,
                                portuguese: None,
                                russian: None,
                                german: None,
                                korean: None,
                                japanese: None,
                                chinese: None,
                                chinesesimp: None,
                                turkish: None,
                                swedish: None,
                                slovak: None,
                                serbocroatian: None,
                                norwegian: None,
                                icelandic: None,
                                hungarian: None,
                                greek: None,
                                finnish: None,
                                dutch: None,
                                ukrainian: None,
                                danish: None,
                            },
                        ],
                        containers: [],
                    },
                ],
            },
        ],
    },
    path: "/containers.xml",
    keys: {
        "a1": [
            Position {
                start: LineCol(
                    105,
                    (
                        4,
                        18,
                    ),
                ),
                end: LineCol(
                    107,
                    (
                        4,
                        20,
                    ),
                ),
                path: "/containers.xml",
            },
        ],
        "b1": [
            Position {
                start: LineCol(
                    215,
                    (
                        8,
                        22,
                    ),
                ),
                end: LineCol(
                    217,
                    (
                        8,
                        24,
                    ),
                ),
                path: "/containers.xml",
            },
        ],
        "a2": [
            Position {
                start: LineCol(
                    304,
                    (
                        11,
                        22,
                    ),
                ),
                end: LineCol(
                    306,
                    (
                        11,
                        24,
                    ),
                ),
                path: "/containers.xml",
            },
        ],
        "a3": [
            Position {
                start: LineCol(
                    473,
                    (
                        17,
                        22,
                    ),
                ),
                end: LineCol(
                    475,
                    (
                        17,
                        24,
                    ),
                ),
                path: "/containers.xml",
            },
        ],
    },
    source: "<?xml version=\"1.0\" encoding=\"utf-8\" ?>\n<Project name=\"test\">\n    <Package name=\"test\">\n        <Key ID=\"A1\">\n            <Original>Origin</Original>\n        </Key>\n        <Container name=\"B\">\n            <Key ID=\"B1\">\n                <Original>Origin</Original>\n            </Key>\n            <Key ID=\"A2\">\n                <German>DE</German>\n                <English>EN</English>\n            </Key>\n        </Container>\n        <Container name=\"A\">\n            <Key ID=\"A3\">\n                <Original>Origin</Original>\n            </Key>\n        </Container>\n    </Package>\n</Project>\n",
    comments: [],
}
